purchases/src/test/java/com/revenuecat/purchases/caching/InMemoryCachedObjectTest.kt

```diff
@@ -0,0 +1,103 @@
+package com.revenuecat.purchases.caching
+
+import com.revenuecat.purchases.Offerings
+import com.revenuecat.purchases.utils.DateTimeProvider
+import io.mockk.mockk
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import java.util.Date
+
+class InMemoryCachedObjectTest {
+    private val epoch = Date(0)
+
+
+    // region isCacheStale
+
+    @Test
+    fun `cache is not set`() {
+        val inMemoryCachedObject = InMemoryCachedObject<Offerings>(1)
+        assertThat(inMemoryCachedObject.isCacheStale()).isTrue()
+    }
+
+    @Test
+    fun `cache is stale`() {
+        val inMemoryCachedObject = InMemoryCachedObject<Offerings>(1)
+        inMemoryCachedObject.updateCacheTimestamp(epoch)
+        assertThat(inMemoryCachedObject.isCacheStale()).isTrue()
+    }
+
+    @Test
+    fun `cache is not stale`() {
+        val inMemoryCachedObject = InMemoryCachedObject<Offerings>(100)
+        inMemoryCachedObject.updateCacheTimestamp(Date())
+        assertThat(inMemoryCachedObject.isCacheStale()).isFalse()
+    }
+
+    // endregion
+
+    // region clearCacheTimestamp
+
+    @Test
+    fun `cache timestamp is cleared correctly`() {
+        val inMemoryCachedObject = InMemoryCachedObject<Offerings>(100)
+        val date = Date()
+        inMemoryCachedObject.updateCacheTimestamp(date)
+        assertThat(inMemoryCachedObject.lastUpdatedAt).isEqualTo(date)
+        inMemoryCachedObject.clearCacheTimestamp()
+        assertThat(inMemoryCachedObject.lastUpdatedAt).isNull()
+    }
+
+    // endregion
+
+    // region clearCache
+
+    @Test
+    fun `cache is cleared correctly`() {
+        val inMemoryCachedObject = InMemoryCachedObject<Offerings>(100)
+        assertThat(inMemoryCachedObject.cachedInstance).isNull()
+        inMemoryCachedObject.cacheInstance(mockk())
+        inMemoryCachedObject.clearCache()
```

Comment:
I'd add a line right before this one to check that after caching it wasn't null, otherwise if caching didn't work at all this test would pass. 
```
assertThat(inMemoryCachedObject.cachedInstance).isNotNull()
inMemoryCachedObject.clearCache()
```

Reply:
then again, you already have a test for that, so it might be redundant, so FFTI